internal/store/serverless_instances.go

```diff
@@ -19,42 +19,60 @@ import (
 
 	"github.com/mongodb/mongodb-atlas-cli/internal/config"
 	atlas "go.mongodb.org/atlas/mongodbatlas"
+	atlasv2 "go.mongodb.org/atlas/mongodbatlasv2"
 )
 
 //go:generate mockgen -destination=../mocks/mock_serverless_instances.go -package=mocks github.com/mongodb/mongodb-atlas-cli/internal/store ServerlessInstanceLister,ServerlessInstanceDescriber,ServerlessInstanceDeleter,ServerlessInstanceCreator,ServerlessInstanceUpdater
 
 type ServerlessInstanceLister interface {
-	ServerlessInstances(string, *atlas.ListOptions) (*atlas.ClustersResponse, error)
+	ServerlessInstances(string, *atlas.ListOptions) ([]atlasv2.ServerlessInstanceDescription, error)
 }
 
 type ServerlessInstanceDescriber interface {
 	ServerlessInstance(string, string) (*atlas.Cluster, error)
+	GetServerlessInstance(string, string) (*atlasv2.ServerlessInstanceDescription, error)
 }
 
 type ServerlessInstanceDeleter interface {
 	DeleteServerlessInstance(string, string) error
 }
 
 type ServerlessInstanceCreator interface {
-	CreateServerlessInstance(string, *atlas.ServerlessCreateRequestParams) (*atlas.Cluster, error)
+	CreateServerlessInstance(string, *atlasv2.ServerlessInstanceDescriptionCreate) (*atlasv2.ServerlessInstanceDescription, error)
 }
 
 type ServerlessInstanceUpdater interface {
-	UpdateServerlessInstance(string, string, *atlas.ServerlessUpdateRequestParams) (*atlas.Cluster, error)
+	UpdateServerlessInstance(string, string, *atlasv2.ServerlessInstanceDescriptionUpdate) (*atlasv2.ServerlessInstanceDescription, error)
 }
 
 // ServerlessInstances encapsulates the logic to manage different cloud providers.
-func (s *Store) ServerlessInstances(projectID string, listOps *atlas.ListOptions) (*atlas.ClustersResponse, error) {
+func (s *Store) ServerlessInstances(projectID string, listOps *atlas.ListOptions) ([]atlasv2.ServerlessInstanceDescription, error) {
 	switch s.service {
 	case config.CloudService:
-		result, _, err := s.client.(*atlas.Client).ServerlessInstances.List(s.ctx, projectID, listOps)
-		return result, err
+		result, _, err := s.clientv2.ServerlessInstancesApi.ListServerlessInstances(s.ctx, projectID).
+			IncludeCount(listOps.IncludeCount).
```

Comment:
we don't expose `IncludeCount` in the cli and assume this to always be true, be sure this is not defaulting to false and the behavior of this default is a bit spotty

Reply:
>Would make sense for us to apply default: true on openapi leve

Should already be
<img width="845" alt="Screenshot 2023-04-20 at 17 00 14" src="https://user-images.githubusercontent.com/461027/233422542-7fcd4f0e-e3d6-4a51-a893-99781e010e13.png">

But then we had this PR https://github.com/mongodb/go-client-mongodb-atlas/pull/443 becuase it seems like the generated client was not always respecting defaults

> Which is weird as in the link it looks like includeCount is set to false...

@wtrocki may be related to the above PR?

Reply:
@andmatei `./bin/atlas serverless list -P dev -o=json --debug` would show you what was the url present.

When we do not include value (it is not set in the struct) we will set it to default value. 
That is why we would not need include count present in any of the stores. 
https://github.com/mongodb/go-client-mongodb-atlas/blob/68e7e21de87ddc93a4c6aae9943f40a0891cbe1a/mongodbatlasv2/api_serverless_instances.go#L557-L558

Reply:
It is set to default value but never present in URL. Fixing in latest dev. Good we caught that.

Reply:
Pushed commit. It should be working now